app/controllers/two_factor_auths_controller.rb

```diff
@@ -0,0 +1,24 @@
+class TwoFactorAuthsController < ApplicationController
+  skip_before_action :validate_two_factor_key
+  layout "login"
+
+  def new
+  end
+
+  def create
+    code = params[:two_factor_auth][:code]
```

Comment:
You could consider using strong parameters here. Just to make sure it's only `two_factor_auth[code]` that gets submitted.